homeassistant/components/sensor/swiss_public_transport.py

```diff
@@ -0,0 +1,148 @@
+"""
+homeassistant.components.sensor.swiss_public_transport
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Swiss public transport sensor will give you the next two departure times
+from a given location to another one. This sensor is limited to Switzerland.
+
+Configuration:
+
+To use the Swiss public transport sensor you will need to add something like
+the following to your config/configuration.yaml
+
+sensor:
+  platform: swiss_public_transport
+  from: STATION_ID
+  to: STATION_ID
+
+Variables:
+
+from
+*Required
+Start station/stop of your trip. To search for the ID of the station, use the
+an URL like this: http://transport.opendata.ch/v1/locations?query=Wankdorf
+to query for the station. If the score is 100 ("score":"100" in the response),
+it is a perfect match.
+
+to
+*Required
+Destination station/stop of the trip. Same procedure as for the start station.
+
+Details for the API : http://transport.opendata.ch
+"""
+import logging
+from datetime import timedelta
+
+from homeassistant.util import Throttle
+import homeassistant.util.dt as dt_util
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+_RESOURCE = 'http://transport.opendata.ch/v1/'
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Get the Swiss public transport sensor. """
+
+    if None in (hass.config.latitude, hass.config.longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+        return False
+
+    try:
+        # pylint: disable=unused-variable
+        from requests import get
```

Comment:
Don't worry about this dependency, requests is a core dependency of Home Assistant and always available